app/javascript/components/challenges/Login.tsx

```diff
@@ -19,6 +25,22 @@ export default function Login() {
 
   const authToken = document.querySelector('head meta[name="csrf-token"]' as any).content;
 
+  // toggle to show/hide password
+  const [showP, setShowP] = React.useState<boolean>(false);
```

Comment:
Again a minor point but don't abbreviate variables makes it more difficult for the next person